packages/utils/telemetry-utils/src/utils.ts

```diff
@@ -69,31 +69,43 @@ export function createSampledLogger(
 			// 2. If isSamplingDisabled is false, then event should be sampled using the event sampler, if the sampler is not defined just send all events, other use the eventSampler.sample() method.
 			// 3. If skipLoggingWhenSamplingIsDisabled is true, then no event is sent.
 			if (isSamplingDisabled || eventSampler === undefined || eventSampler.sample()) {
-				if (isSamplingDisabled && skipLoggingWhenSamplingIsDisabled) {
+				if (isSamplingDisabled && (skipLoggingWhenSamplingIsDisabled ?? false)) {
 					return;
 				}
 				logger.send(event);
 			}
 		},
 		sendTelemetryEvent: (event: ITelemetryGenericEventExt): void => {
-			if (isSamplingDisabled || eventSampler === undefined || eventSampler.sample()) {
-				if (isSamplingDisabled && skipLoggingWhenSamplingIsDisabled) {
+			if (
+				isSamplingDisabled ||
+				eventSampler === undefined ||
+				(eventSampler.sample() ?? false)
```

Comment:
I think the build should now complain about the unnecessary coalescing left over in 3 places?

Reply:
I don't think it did locally. Let's see what happens in CI.

Reply:
Didn't fail in CI, but I'll fix these regardless.